src/packages/core/src/resolvers.ts

```diff
@@ -686,6 +692,44 @@ const _listRelationshipField = async <G, D, R, C extends BaseContext>(
 
 	logger.trace('Loading from BaseLoaders');
 
+	let relationshipFilterChunkFound = !relationshipFilterChunk;
+	/**
+	 * Look for `relationshipFilterChunk` and remove it from the filter.
+	 * This is necessary because we are batch loading data and that relationshipFilterChunk makes the filter different for each record, which ends up triggering a SQL query for each record (not batching at all).
+	 * What we want to achieve is having a filter with ACLs into it, but without the relationship chunk.
+	 */
+	const removeRelationshipFilterChunk = (
```

Comment:
Is there any reason to define this as an inline function instance? Could you just make it a statement that is calculated and assigned to `loadFilter`?

Reply:
It's a recursive function